tests/test_scm_record.rs

```diff
@@ -2853,3 +2853,111 @@ fn test_no_files() -> eyre::Result<()> {
+                            line: Cow::Borrowed("\tbefore text\t4\n"),
+                        },
+                        SectionChangedLine {
+                            is_checked: true,
+                            change_type: ChangeType::Added,
+                            line: Cow::Borrowed("\tafter text 4\n"),
+                        },
+                        SectionChangedLine {
+                            is_checked: true,
+                            change_type: ChangeType::Removed,
+                            line: Cow::Borrowed("\tbefore text\t5"),
+                        },
+                        SectionChangedLine {
+                            is_checked: true,
+                            change_type: ChangeType::Added,
+                            line: Cow::Borrowed("\tafter text\t5"),
+                        },
+                    ],
+                },
+                Section::Unchanged {
+                    lines: vec![Cow::Borrowed("this is some trailing\ttext\n")],
+                },
+            ],
+        }],
+    };
+    let initial = TestingScreenshot::default();
+    let mut input = TestingInput::new(
+        80,
+        18,
+        [Event::ExpandAll, initial.event(), Event::QuitAccept],
+    );
+    let recorder = Recorder::new(state, &mut input);
+    recorder.run()?;
+
+    insta::assert_snapshot!(initial, @r###"
+    "[File] [Edit] [Select] [View]                                                   "
+    "(×) foo/bar                                                                  (-)"
+    "        ⋮                                                                       "
+    "        8 →   this is some indented text⏎                                       "
+    "        9 →   this is some indented text⏎                                       "
+    "       10 →   this is some indented text⏎                                       "
+    "  [×] Section 1/1                                                            [-]"
+    "    [×] - before text→   1⏎                                                     "
+    "    [×] + after text 1⏎                                                         "
+    "    [×] - before text 2⏎                                                        "
+    "    [×] + after text→   2⏎                                                      "
+    "    [×] - →   before text 3⏎                                                    "
+    "    [×] + →   after text→   3⏎                                                  "
+    "    [×] - →   before text→   4⏎                                                 "
+    "    [×] + →   after text 4⏎                                                     "
+    "    [×] - →   before text→   5                                                  "
+    "    [×] + →   after text→   5                                                   "
+    "       16 this is some trailing→   text⏎                                        "
+    "###);
```

Comment:
note: I'm really not sure if this is a worthwhile test, or what changes should be made, but it _does_ fail if https://github.com/arxanas/scm-record/pull/37/files#diff-acf9ca25b7b2d13e11c1035ebbda4140466603739b73a371b67a9a267d0ce626R3055 is removed.

Reply:
It seems like a fine test to me. Squash it into the commit that adds the corresponding behavior so that none of the commits in this stack fail tests.

Reply:
Done in [`000a234` (#37)](https://github.com/arxanas/scm-record/pull/37/commits/000a2349684370866b7da005db3c6d8936a8e26f)